iree/compiler/Conversion/LinalgToSPIRV/LinalgTileAndFusePass.cpp

```diff
@@ -488,10 +539,54 @@ void LinalgTileAndFusePass::runOnOperation() {
         });
       }
 
+      // For the next step, we will need to query the workgroup count function
+      // to simplify GPU processor ID uses. It relies on constant upper bounds.
+      // So we need to canonicalize the workgroup count function first.
+      if (funcOp.getAttrOfType<SymbolRefAttr>(getNumWorkgroupsFnAttrName())) {
```

Comment:
Maybe move this up right after tiling to workgroups? Also move this into its own function.